source/src/cip/cipcommon.c

```diff
@@ -866,6 +866,10 @@ void AllocateAttributeMasks(CipClass *target_class) {
 
 size_t CalculateIndex(EipUint16 attribute_number) {
   size_t index = (attribute_number + 8 - 1) / 8;
+  OPENER_ASSERT(index>0);
```

Comment:
Hi! If this assert is not fulfilled, the if will not be needed. I know, only as long as #define NDEBUG 
 is not defined, but I usually use assert as a guarantee, which has been tested in debug and should hold on release. If you think the check is needed on release, than a trace message should be added

Reply:
Not working for the conformance tester. At requesting attribute 0 the assert cannot be fulfilled

Reply:
Attribute 0 is legal?! I didn't realize that. I agree, this patch is broken for attribute 0. Thing is, there's lots of "1 << ( (attribute_number - 1) % 8 )" code converting from attribute number to the bit position that may also be a problem with attribute 0. Hmm. 

Of the top of my head, I'm thinking ((attribute_number / 8) + 1) for the array size, (attribute_number / 8) for the index, and (1 << (attribute_number %8)) for the bit position. Does this make sense?